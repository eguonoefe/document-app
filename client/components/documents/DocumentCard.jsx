import React from 'react';
import PropTypes from 'prop-types';
import parseDate from '../../utilities/parseDate';

/**
* Document Card for viewing a single document
*
* @param {Object} props { title, id, access, updatedAt, onClick, User }
* @returns {String} HTML markup for the Document Card
*/
const DocumentCard = (props) => {
  const { title, id, access, updatedAt, onClick, User } = props;
  return (
    <div>
      <a
        onClick={onClick}
        name={id}
        className="truncate"
        href="#!">{title}</a>
      <p>By {User.firstName} {User.lastName}</p>
      {(access === 'public') && <p>Public</p>}
      {(access === 'private') && <p>Private</p>}
      {(access === 'role') && <p>Role</p>}
      <p>Last Edited: {parseDate(updatedAt)}</p>
      <div className="divider" />
    </div>
  );
};

// Set Props
DocumentCard.propTypes = {
  title: PropTypes.string,
  access: PropTypes.string,
  id: PropTypes.number,
  User: PropTypes.object,
  onClick: PropTypes.func,
  updatedAt: PropTypes.node
};

export default DocumentCard;
