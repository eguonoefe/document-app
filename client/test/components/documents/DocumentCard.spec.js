/* global expect jest test */
import React from 'react';
import { shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import DocumentCard from '../../../components/documents/DocumentCard';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/documents/DocumentCard');

describe('DocumentCard Component', () => {
  test('should match the document card snapshot', () => {
    const onClick = jest.fn();
    const User = {
      firstName: '',
      lastName: ''
    };
    const component = shallow(
      <DocumentCard
        title={'Book'}
        id={3}
        access={'public'}
        updatedAt={'today'}
        onClick={onClick}
        User={User} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
