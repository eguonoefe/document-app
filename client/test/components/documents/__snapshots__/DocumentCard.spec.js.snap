// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentCard Component should match the document card snapshot 1`] = `
<div>
  <a
    className="truncate"
    href="#!"
    name={3}
    onClick={[MockFunction]}
  >
    Book
  </a>
  <p>
    By 
     
  </p>
  <p>
    Public
  </p>
  <p>
    Last Edited: 
    Invalid Date
  </p>
  <div
    className="divider"
  />
</div>
`;
